src/Symfony/Component/Translation/Translator.php

```diff
@@ -193,12 +193,20 @@ public function getFallbackLocales(): array
     /**
      * {@inheritdoc}
      */
-    public function trans(?string $id, array $parameters = [], string $domain = null, string $locale = null)
+    public function trans($id, array $parameters = [], string $domain = null, string $locale = null)
     {
+        if ($id instanceof Translatable) {
+            $parameters += $id->getParameters();
+            $domain = $id->getDomain();
+            $id = $id->getMessageId();
+        }
+
         if (null === $id || '' === $id) {
```

Comment:
i woudl keep this "if" the first block, for clarity

Reply:
No longer needed